collect_app/src/main/java/org/odk/collect/android/widgets/BearingWidget.java

```diff
@@ -21,176 +21,126 @@
-        this.answer.setText((String) answer);
+        binding.answerText.setText((String) answer);
         widgetValueChanged();
     }
 
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
         if (isSensorAvailable) {
-            getBearingButton.setOnLongClickListener(l);
+            binding.bearingButton.setOnLongClickListener(l);
         }
-        answer.setOnLongClickListener(l);
+        binding.answerText.setOnLongClickListener(l);
     }
 
     @Override
     public void cancelLongPress() {
         super.cancelLongPress();
         if (isSensorAvailable) {
-            getBearingButton.cancelLongPress();
+            binding.bearingButton.cancelLongPress();
         }
-        answer.cancelLongPress();
+        binding.answerText.cancelLongPress();
     }
 
     private boolean checkForRequiredSensors() {
-
         boolean isAccelerometerSensorAvailable = false;
         boolean isMagneticFieldSensorAvailable = false;
 
-        SensorManager sensorManager = (SensorManager) getContext().getSystemService(Context.SENSOR_SERVICE);
         if (sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER) != null) {
             isAccelerometerSensorAvailable = true;
         }
         if (sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD) != null) {
             isMagneticFieldSensorAvailable = true;
         }
-
         return isAccelerometerSensorAvailable && isMagneticFieldSensorAvailable;
     }
 
-    private EditText getEditText() {
-        final EditText manualData = new EditText(getContext());
-        manualData.setPadding(20, 20, 20, 20);
-        manualData.setTextSize(TypedValue.COMPLEX_UNIT_DIP, getAnswerFontSize());
-        manualData.setInputType(InputType.TYPE_NUMBER_FLAG_DECIMAL);
-        manualData.setKeyListener(new DigitsKeyListener(true, true));
-        manualData.setFocusable(false);
-        manualData.setFocusableInTouchMode(false);
-        TableLayout.LayoutParams params = new TableLayout.LayoutParams();
-        params.setMargins(7, 5, 7, 5);
-        manualData.setLayoutParams(params);
-
-        manualData.addTextChangedListener(new TextWatcher() {
-            @Override
-            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
-
-            }
-
-            @Override
-            public void onTextChanged(CharSequence s, int start, int before, int count) {
-
-            }
-
-            @Override
-            public void afterTextChanged(Editable s) {
-                widgetValueChanged();
-            }
-        });
-
-        return manualData;
-    }
-
-    @Override
-    public void onButtonClick(int buttonId) {
+    private void onButtonClick() {
         if (isSensorAvailable) {
-            Intent i = new Intent(getContext(), BearingActivity.class);
+            Intent intent = new Intent(getContext(), BearingActivity.class);
             waitingForDataRegistry.waitForData(getFormEntryPrompt().getIndex());
```

Comment:
Could you move it one line down? The reason is like below (mixing stuff).

Reply:
Something went wrong here because you just added and empty line instead of moving the line I mentioned down.